lib/grape/endpoint.rb

```diff
@@ -210,47 +216,49 @@ def equals?(e)
     protected
 
     def run(env)
-      @env = env
-      @header = {}
+      ActiveSupport::Notifications.instrument('run.grape', endpoint: self, env: env) do
+        @env = env
```

Comment:
Lets refactor the body into a method? Maybe `run` should call `instrumented_run` or something like that?